spec/govuk_error/configuration_spec.rb

```diff
@@ -27,14 +27,27 @@
     it "ignores errors if they happen in an environment we don't care about" do
       ClimateControl.modify SENTRY_CURRENT_ENV: "some-temporary-environment" do
         configuration.active_sentry_environments << "production"
-        expect(configuration.before_send.call(StandardError.new)).to be_nil
+        sentry_client = send_exception_to_sentry(StandardError.new, configuration)
+        expect(sentry_client.transport.events).to be_empty
       end
     end
 
     it "captures errors if they happen in an environment we care about" do
       ClimateControl.modify SENTRY_CURRENT_ENV: "production" do
         configuration.active_sentry_environments << "production"
-        expect(configuration.before_send.call(StandardError.new)).to be_truthy
+        sentry_client = send_exception_to_sentry(StandardError.new, configuration)
+        expect(sentry_client.transport.events.count).to eq(1)
+      end
+    end
+
+    it "allows string messages to be sent (rather than exceptions)" do
```

Comment:
Thanks for adding this test and your explanation on Slack that `Sentry::Utils::ExceptionCauseChain.exception_to_array()` can cope with an input of nil. One thing I noticed in trying that out was that this test doesn't execute https://github.com/alphagov/govuk_app_config/blob/2f377a28eb37983e14a4e4918e54b27fc8b9187e/lib/govuk_app_config/govuk_error/configuration.rb#L36-L42

since data_sync_excluded_exceptions is an empty array. I think it could be useful to have that covered in the test case as passing that method nil seems a risky integration point.

Reply:
Good spot. I've now populated the array.